torch_xla/csrc/xla_graph_executor.h

```diff
@@ -209,12 +209,21 @@ class XLAGraphExecutor : public torch::lazy::LazyGraphExecutor {
         const std::vector<size_t>* indices,
         DebugUtil::GraphFormat format = DebugUtil::GetDefaultGraphFormat());
 
+    void SaveOutputShapes(torch::lazy::hash_t hash,
+                          std::vector<xla::Shape> outptu_shapes);
```

Comment:
`outptu_shapes` => `output_shapes`, and `const std::vector<xla::Shape>&` perhaps?

Reply:
`outptu_shapes` will be saved in a global map and outlive the stack object from the caller, I think it has to be a copy.

Reply:
it is also `std::move` so we are not making an extra copy.

Reply:
I guess my hobby has always been explicitly making the copy in the code instead of relying on value passing. That's why I always ask. Otherwise, it's too much thinking on deciding whether a parameter should be passed by value or reference. Just easier to determine if it's const reference or r reference.

Reply:
sg, will update.

Reply:
ah, actually in this case we want to keep it as `std::vector<xla::Shape>`. This way caller can use `std::move` to avoid the extra copy we need to do if we only pass in a reference.

Reply:
Haha, it's tricky. I'm actually fine with either ways. Just trying to share some of my thoughts here.